dessia_common/excel_reader.py

```diff
@@ -0,0 +1,474 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+""" ExcelReader for DessiaObject. """
+
+import inspect
+from ast import literal_eval
+from typing import Dict, List
+
+import openpyxl
+
+from dessia_common.utils.helpers import get_python_class_from_class_name
+
+
+def check_duplicate_attributes(attributes):
+    """
+    Check for the presence of duplicate attributes in a list of attribute names.
+
+    Args:
+    - attributes (list): A list containing attribute names to be checked.
+
+    Returns:
+    - str or None: If duplicates are found, returns the first duplicated attribute name; otherwise, returns None.
+    """
+    seen = set()
+    for attribute in attributes:
+        if attribute in seen:
+            return attribute
+        seen.add(attribute)
+    return None
+
+
+class ExcelDataExtract:
+    """Data extracted from sheet of Excel file."""
+
+    def __init__(self, class_module: str, class_name: str, attributes: List[str], datas):
+        self.class_module = class_module
+        self.class_name = class_name
+        self.attributes = attributes
+        self.datas = datas
+        self.missed_attribute()
+
+    @property
+    def object_class(self) -> str:
+        """Get the full name of the class."""
+        return f"{self.class_module}.{self.class_name}"
+
+    @property
+    def class_(self):
+        """Get the Python class object corresponding to the fully class name."""
+        return get_python_class_from_class_name(self.object_class)
+
+    @property
+    def get_attributes_and_types(self):
+        """
+        Retrieve attributes and their corresponding types from the __init__ method of a class.
+
+        Returns:
+        - dict: A dictionary containing attributes and their respective types defined in the class constructor.
+        """
+        init_signature = inspect.signature(self.class_.__init__)
+        init_parameters = init_signature.parameters
+
+        attribute_info = {}
+
+        for attr_name, param in init_parameters.items():
```

Comment:
You definitely need schemas here. I'll validate the PR so that we can pass it in the upcoming release. We'll change this in a further PR